packages/core/saved-objects/core-saved-objects-base-server-internal/src/validation/validator.ts

```diff
@@ -46,12 +47,15 @@ export class SavedObjectsTypeValidator {
   }
 
   public validate(document: SavedObjectSanitizedDoc, version?: string): void {
-    const docVersion =
-      version ??
-      document.typeMigrationVersion ??
-      document.migrationVersion?.[document.type] ??
-      this.defaultVersion;
-    const schemaVersion = previousVersionWithSchema(this.orderedVersions, docVersion);
+    let usedVersion = version;
```

Comment:
Even though the `version` option is (or should be) unused now, keeping the behavior allows us to reuse it if needed.

I'm open to removing the option if someone objects.

Reply:
Correct me if I'm wrong, but in current state, the `version` option is effectively unused (given `usedVersion` is always replaced L52-L56), right?

I'm fine either way (keeping it or removing it), but if we do keep it, then we need the `version` parameter to be used if specified (it doesn't make sense to keep an option ignored by the implementation), and to adapt the call to `validate` to no longer pass the option, here:

https://github.com/elastic/kibana/blob/3b6b7ad9b9553be3d039c71edcbdcb2e3d6423fd/packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/helpers/validation.ts#L93-L95